server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -1839,12 +1588,46 @@ private long generateQueryId(boolean isDataQuery, int fetchSize, int deduplicate
     return SchemaUtils.getSeriesTypesByPaths(paths, aggregations);
   }
 
-
   protected TSDataType getSeriesTypeByPath(PartialPath path) throws MetadataException {
     return SchemaUtils.getSeriesTypeByPath(path);
   }
 
-  public static int getDefaultFetchSize() {
-    return DEFAULT_FETCH_SIZE;
+  private TSStatus onException(Exception e, String operation) {
+    TSStatus status = tryCatchQueryException(e);
+    return status != null ? status : onNPEOrUnexpectedException(e, operation,
+        TSStatusCode.INTERNAL_SERVER_ERROR);
+  }
+
+  private TSStatus tryCatchQueryException(Exception e) {
```

Comment:
Rename as `onQueryException`?

Reply:
I think the name tryCatchQueryException is better, because errors that flow through the tryCatchQueryException method may not be caught.